src/adaXT/criteria/criteria.pyx

```diff
@@ -541,9 +542,107 @@ cdef class Linear_regression(Criteria):
             int i, length
 
         length = indices.shape[0]
-        theta0, theta1 = self.theta(indices)
+        theta0, theta1 = self._theta(indices)
         cur_sum = 0.0
         for i in range(length):
             step_calc = self.y[indices[i]] - theta0 - theta1 * self.x[indices[i], 0]
-            cur_sum += step_calc*step_calc
+            cur_sum += step_calc ** 2
+        return cur_sum
+
+cdef class Partial_quadratic(Criteria):
```

Comment:
Might be an idea to create a "Partial" parent class, which inherits from the Criteria class. That way the _custom_mean does not have to be created twice.

Reply:
Might be a good idea if we extend further, currently the custom mean is different for both so lets just keep it as is for now.